aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp

```diff
@@ -11,6 +11,197 @@
 
 torch::class_<EmbeddingPackedParamsBase> register_embedding_params();
 
+namespace {
+at::Tensor embedding_bag_4bit_helper(
+    const at::Tensor& weight,
+    const at::Tensor& indices,
+    const c10::optional<at::Tensor>& offsets_in,
+    bool sparse,
+    const c10::optional<at::Tensor>& per_sample_weights_,
+    const c10::optional<at::Tensor>& compressed_indices_mapping,
+    bool include_last_offset) {
+  TORCH_CHECK(
+      offsets_in.has_value(),
+      "embedding_bag_4bit_rowwise_offsets expects offsets to be set");
+
+  TORCH_CHECK(weight.ndimension() == 2);
+  TORCH_CHECK(indices.ndimension() == 1);
+
+  auto offsets = offsets_in.value();
+  TORCH_CHECK(offsets.ndimension() == 1);
+
+  // FBGEMM expects the offsets to be of int type.
+  at::Tensor offsets_new = offsets.toType(at::ScalarType::Int);
+
+  auto offsets_data = offsets_new.data_ptr<int>();
+  const auto weight_data = weight.data_ptr<uint8_t>();
+  uint8_t* input_data = nullptr;
+  if (!weight.is_contiguous()) {
```

Comment:
I think you can just do 
```
auto weight_contig = weight.contiguous();
input_data = weight_contig.data_ptr<uint8_t>();
```
without checking if weight is contigous or not here.